inrupt/pod-browser
# handle no pod URL and error fetching extended profile properly

## Description

Whenever PB cannot find a Pod URL in the profile, it completely hangs. Instead we want to display an error to users. If an extended profile cannot be fetched, we want the function retrieving the profile to ignore this and move on.

This PR introduces a fix to handle the case where a Pod URL cannot be found on a profile, and also fixes a bug where `getFullProfile` was not handling failing to fetch an extended profile dataset properly (ignoring the error and moving on).

I took some design liberties (error messages) so design review is required.

## Testing

Case 1: fail to fetch an extended profile

1. https://id.inrupt.com/ edit your WebID by changing the URLs to the extended profile (highlighted in screenshot below)
<img width="695" alt="image" src="https://user-images.githubusercontent.com/28412960/186151853-e1a5dd6d-8afc-4e11-8a76-cf0ed81dc798.png">
2. Go to the Vercel deployment of this branch (Preview) and verify everything works correctly (main files page, profile).

Case 2: no Pod URL found in profile

1. https://id.inrupt.com/ edit your WebID by deleting the storage triple (highlighted in screenshot below):
<img width="695" alt="image" src="https://user-images.githubusercontent.com/28412960/186152481-71d869dd-0923-4691-af91-e2f04c4b7928.png">

2. Go to the Vercel deployment of this branch (Preview) and verify everything works correctly (main files page, profile).

## Commit checklist
